src/styles/shared/components/Buttons/_index.scss

```diff
@@ -0,0 +1,168 @@
+    'hover-background': transparent,
+    'active-foreground': map.get($-settings, 'color-brand'),
+    'active-background': transparent,
+  )
+);
+
+$IconButton-colors--primary: map.merge(
+  $IconButton-colors,
+  (
+    'foreground': map.get($-settings, 'color-brand'),
+    'hover-foreground': map.get($-settings, 'color-brand'),
+    'active-foreground': map.get($-settings, 'color-brand'),
+  )
+);
+
+// LABELED BUTTON colors
+$LabeledButton-colors: $colors;
+
+// This set of colors is light text on dark background
+$LabeledButton-colors--primary: map.merge(
+  $colors,
+  (
+    'foreground': map.get($-settings, 'color-g1'),
+    'background': map.get($-settings, 'color-gmid'),
+    'hover-foreground': map.get($-settings, 'color-g1'),
+    'hover-background': map.get($-settings, 'color-g6'),
+    'disabled-foreground': map.get($-settings, 'color-gsemi'),
+  )
+);
+
+// This set of colors is for buttons styled as links on a white background
+$LinkButton-colors: map.merge(
+  $colors,
+  (
+    'background': transparent,
+    'hover-background': transparent,
+    'active-background': transparent,
+    'hover-foreground': map.get($-settings, 'color-link-hover'),
+  )
+);
+
+// Public MIXINS ////////////////////////////////
+// Button base block element mixin
+@mixin Button(
+  $colormap: $colors,
+  $settings: $-settings,
+  $withStates: true,
+  $withLayout: false
+) {
+  @include mixins.button(
+    $colormap: $colormap,
+    $settings: $settings,
+    $withStates: $withStates,
+    $withLayout: $withLayout
+  ) {
+    @content;
+  }
+}
+
+// Mixin for --modifier variants on Buttons, e.g. `.MyButton--primary`
+@mixin Button--variant($colormap: $colors, $withStates: false) {
+  @include mixins.variant($colormap: $colormap, $withStates: $withStates);
+}
+
+// Block mixin for a button with an icon and nothing else
+@mixin IconButton {
+  @include Button($colormap: $IconButton-colors, $withStates: true) {
+    @media (pointer: coarse) {
+      min-width: map.get($-settings, 'touch-target-size');
+      min-height: map.get($-settings, 'touch-target-size');
+    }
+  }
+  &--primary {
+    @include Button--variant($IconButton-colors--primary, $withStates: true);
+  }
+  @content;
+}
+
+// Block mixin for a button with a text label and (optional) icon
+@mixin LabeledButton {
+  @include Button(
+    $colormap: $LabeledButton-colors,
+    $withLayout: true,
+    $withStates: true
+  );
+  &--primary {
+    @include Button--variant($LabeledButton-colors--primary, $withStates: true);
+  }
+  @content;
+}
+
+// Block mixin for a button styled to look like a link
+@mixin LinkButton {
+  @include Button($LinkButton-colors, $withStates: false) {
+    // Override base font-weight
+    font-weight: 400;
+  }
+  // No active states
+  @include mixins.hover-state($LinkButton-colors) {
+    text-decoration: underline;
+  }
+  // Remove horizontal padding to allow button to appear flush-left or -right
+  padding: 0.5em 0;
+  @content;
+}
+
+@mixin styles {
```

Comment:
I think calling this mixin `components` rather than `styles` would have made it more obvious to me what was going on in the index file.